src/main/java/uk/nhs/digital/ps/migrator/model/hippo/XlsxReader.java

```diff
@@ -0,0 +1,109 @@
+package uk.nhs.digital.ps.migrator.model.hippo;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.AbstractMap;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+/**
+ * Shared functions for streaming and reading Excel sheets/rows.
+ */
+public class XlsxReader {
+ 
+    private Map<String, Integer> headerNameIndexPairs;
+
+    /**
+     * Initialise the reader. This processes the headers and returns the renaming iterator. Headers are 'cleaned' for improved robustness.
+     */
+    public Iterator<Row> init(Path path, final String sheetName){
```

Comment:
Since this method does similar job to the `getRowIterator`, how about reflecting this in its name, say, renaming it to `initAndRenamingRowIterator` (wordy, I know ;-))?

I do recognise that this is not a very fortunate name (especially the 'And' component) but this is a consequence of making this class merging two modes of operation (stateful and stateless). Calling it just `getRenamingRowIterator` would suggest to the reader that this method does not change the state of this object (as expected of all 'getters') and so it makes sense to leave the 'init' component in.

In long lived, production code I'd recommend refactoring the whole class (probably into more than one) to better manage and reflect the stateless vs. stateful methods, but I'm not sure it's worth doing in this application, given its expected short lifespan and relatively small size.